src/test/java/uk/gov/hmcts/reform/migration/service/ApplicantDataMigrationServiceTest.java

```diff
@@ -0,0 +1,229 @@
+import uk.gov.hmcts.reform.domain.OldApplicant;
+import uk.gov.hmcts.reform.domain.common.Address;
+import uk.gov.hmcts.reform.domain.common.EmailAddress;
+import uk.gov.hmcts.reform.domain.common.MobileNumber;
+import uk.gov.hmcts.reform.domain.common.ApplicantParty;
+import uk.gov.hmcts.reform.domain.common.TelephoneNumber;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ApplicantDataMigrationServiceTest {
+    private final String APPLICANT = "Applicant name";
+    private final String EMAIL = "email@email.com";
+    private final String MOBILE = "07778987656";
+    private final String COUNTY = "County";
+    private final String COUNTRY = "Country";
+    private final String POSTCODE = "BT17NT";
+    private final String POSTTOWN = "Ballymena";
+    private final String ADDRESSLINE1 ="ADDRESSLINE1";
+    private final String ADDRESSLINE2 = "ADDRESSLINE2";
+    private final String ADDRESSLINE3 = "ADDRESSLINE3";
+    private final String JOBTITLE = "JobTitle";
+    private final String TELEPHONE = "02825674837";
+    private final String PERSONTOCONTACT = "Person to contact";
+    private final String PARTYTYPE = "ApplicantParty type";
+    private final String LEADAPPLICANTINDICATOR = "Yes";
+    private final String PARTYID = UUID.randomUUID().toString();
+    private final String PBANUMBER= "PBA123456";
+
+    private final ApplicantDataMigrationService service = new ApplicantDataMigrationService();
+
+    @Test
+    void whenOldStructureDoesNotExistAcceptsShouldReturnFalse() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(ImmutableMap.of("data", "someData"))
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(false);
+    }
+    @Test
+    void whenDataIsNullAcceptsShouldReturnFalse() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(null)
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(false);
+    }
+
+    @Test
+    void whenOldApplicantStructureIsMigratedShouldReturnNewAplicantStructure() {
+        Map<String, Object> data = new HashMap<>();
+
+        OldApplicant applicant = getOldApplicant();
+
+        Applicant newApplicant = newApplicantBuilder();
+
+        data.put("applicant", applicant);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(data)
+            .build();
+
+        service.migrate(caseDetails);
+
+
+        List<Map<String, Object>> valueInApplicant = (List<Map<String, Object>>) caseDetails.getData().get("applicants");
+
+        assertThat(valueInApplicant.get(0).get("value").equals(newApplicant));
+    }
+
+    @Test
+    void whenPartiallyFilledInOldApplicantStructureIsMigratedShouldReturnNewListStructureWithNullFields() {
```

Comment:
None of the tests make sure `applicant` is set to null which is an important part of migration.

Reply:
👍